icicle/include/icicle/device.h

```diff
@@ -1,6 +1,8 @@
 #pragma once
 
 #include <cstring>
+#include <string>
```

Comment:
did you change that for Hash

Reply:
I didn't have the includes which is a mistake. It works if you include them before including device.h but it's still a mistake to rely on previous includes.